### DISCUSSION TITLE
[Question][Hw1] Abount undef value of LLVM-IR

### DISCUSSION THREAD

**Discussion by kohs100** [prosyslab-classroom/cs424-program-reasoning#389] [Q&A]
<img width="1108" alt="image" src="https://user-images.githubusercontent.com/20630805/190317517-0c29d513-2492-48fe-961d-9a5307a2d0a7.png">

I found that given match cases in eval function cannot support undef value of LLVM IR.
(I have no idea why compiler creates such code even j is properly initialized at the point it should be)
Is this kind of IR considered as syntactically invalid? Should I add support for undef value, such as initializing with zero?

**Comment by sujin0529** [comment]
The reason why `undef` occurred seems to be because the value of `j` is not initialized in the first `for.cond`.
There is no such problem in our test case.
So, you can ignore this problem!

Thanks.
